pkg/scraper/client/resource/client.go

```diff
@@ -77,18 +82,23 @@ func newClient(c *http.Client, resolver utils.NodeAddressResolver, defaultPort i
 // GetMetrics implements client.KubeletMetricsGetter
 func (kc *kubeletClient) GetMetrics(ctx context.Context, node *corev1.Node) (*storage.MetricsBatch, error) {
 	port := kc.defaultPort
+	path := "/metrics/resource"
 	nodeStatusPort := int(node.Status.DaemonEndpoints.KubeletEndpoint.Port)
 	if kc.useNodeStatusPort && nodeStatusPort != 0 {
 		port = nodeStatusPort
 	}
+	if node.Annotations != nil &&
```

Comment:
nit: even if a map is nil, m["xyz"] will return "" instead of panic:

```suggestion
	if node.Annotations[AnnotationResourceMetricsPath] != "" {
```